### PR TITLE
Feature catalog

### PR DESCRIPTION
**Pull request by iakov-aws**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cid/builtin/core/data/resources.yaml [hunk 1]
```diff
@@ -2,7 +2,9 @@
 # QuickSight Dashboards definitions
 dashboards:
   CUDOS:
+    category: 'Foundational'
     name: CUDOS Dashboard
+    deprecationNotice: ' DEPRECATED. Replaced with CUDOS Dashboard v5.'
```
Comment (yprikhodko) [inline cid/builtin/core/data/resources.yaml:7]: Maybe let's give this param more general purpose name like description?

Comment (iakov-aws) [reply cid/builtin/core/data/resources.yaml:7]: Description can be a separate feature. deprecationNotice is used to detect if it is deprecated or not

File: cid/common.py [hunk 2]
```diff
@@ -361,16 +390,32 @@ def _deploy(self, dashboard_id: str=None, recursive=True, update=False, **kwargs
...
+            standard_categories = ['Foundational', 'Advanced', 'Additional']
+            all_categories = set([f"{dashboard.get('category', 'Custom')}" for dashboard in self.resources.get('dashboards').values()])
+            non_standard_categories = [cat for cat in all_categories if cat not in standard_categories]
+            categories =  standard_categories + sorted(non_standard_categories)
+            dashboard_options = {}
+            for category in categories:
+                dashboard_options[f'{category.upper()}'] = '[category]'
+                for dashboard in self.resources.get('dashboards').values():
+                    if dashboard.get('category', 'Custom') == category:
+                        if dashboard.get('deprecationNotice'):
```
Comment (yprikhodko) [inline cid/common.py:404]: why is this needed?

File: cid/common.py [hunk 3]
```diff
@@ -361,16 +390,32 @@ def _deploy(self, dashboard_id: str=None, recursive=True, update=False, **kwargs
         # TODO: check if datasets returns explicit permission denied and only then discover dashboards as a workaround
         self.qs.discover_dashboards()
 
+        dashboard_id = dashboard_id or get_parameters().get('dashboard-id')
+        if not dashboard_id:
+            standard_categories = ['Foundational', 'Advanced', 'Additional']
```
Comment (yprikhodko) [inline cid/common.py:395]: Why it's hardcoded?

File: cid/common.py [hunk 4]
```diff
@@ -361,16 +390,32 @@ def _deploy(self, dashboard_id: str=None, recursive=True, update=False, **kwargs
...
+            all_categories = set([f"{dashboard.get('category', 'Custom')}" for dashboard in self.resources.get('dashboards').values()])
+            non_standard_categories = [cat for cat in all_categories if cat not in standard_categories]
+            categories =  standard_categories + sorted(non_standard_categories)
+            dashboard_options = {}
+            for category in categories:
+                dashboard_options[f'{category.upper()}'] = '[category]'
+                for dashboard in self.resources.get('dashboards').values():
+                    if dashboard.get('category', 'Custom') == category:
```
Comment (yprikhodko) [inline cid/common.py:403]: Let's change to Other or Uncategorized. Custom has additional context which might confuse customers

**Review by yprikhodko — APPROVED** [review]
